### PR TITLE
Embedded datafiles and arbitrary output path

### PR DESCRIPTION
**Pull request by lierdakil**
Let's try this again. This time, no library. Just embedded datafiles to make binary relocatable and ability to output to arbitrary filepath, including stdout.

Please at least respond, it's rather disheartening when PRs go stale due to almost complete lack of reaction from author/maintainer.

### CODE REVIEW

**Comment by jaspervdj** [comment]
This looks good, and my apologies for being unresponsive before.  I have a few small suggestions which I'll put in line comments.

File: profiteur.cabal [hunk 1]
```diff
@@ -31,6 +31,10 @@ Data-files:
   data/js/unicode.js
   data/js/zoom.js
 
+Flag embed_data
+  description: Embed data files into the executable (needed for ghcjs packaging)
+  default:     False
```
Comment (jaspervdj) [inline profiteur.cabal]: Can we rename this to `embed-data-files` and add `manual: True`?

File: src/Profiteur/DataType.hs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module Profiteur.DataType where
+
+import Data.String
+
+data DataType = JQuery | DataFile FilePath deriving (Show)
```
Comment (jaspervdj) [inline src/Profiteur/DataType.hs]: Can we call this datatype `DataFile` and use `JQueryFile` for the current `JQuery`?

File: src/Profiteur/DataType.hs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module Profiteur.DataType where
```
Comment (jaspervdj) [inline src/Profiteur/DataType.hs]: Can we call this module `Profiteur.DataFile.Internal`?

File: src/embed/Profiteur/Data.hs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Profiteur.Data where
```
Comment (jaspervdj) [inline src/embed/Profiteur/Data.hs]: This module should be called `Profiteur.DataFile` and it should re-export `Profiteur.DataFile.Internal`.  The same goes for the other `Profiteur.DataFile` module.

**Comment by lierdakil** [comment]
Okay, I think I made all the changes you requested. Thanks for reviewing!

**Pull request merged**